### ISSUE TITLE
多轮对话微调如何组织数据格式？

### ISSUE THREAD

**Issue by cristianohello** [THUDM/ChatGLM-6B#410]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

你好，如果领域多轮对话数据的话，如何构建多轮qa 数据示例进行微调？

比如：我有个三轮的对话数据，下面这样构建微调数据集合理吗？q：代表用户说的话，a：代表机器说的话。
{"id": 1, "paragraph": [
{"q": "长城h3风扇不转。继电器好的。保险丝好的传感器新的风扇也新的这是为什么。就是继电器缺一个信号线", "a": ["用电脑能读数据流吗？水温多少"]}
,{"q": "95", "a": ["继续发动水温就高了？", "上下水管温差怎么样啊？空气是不是都排干净了呢？"]}
,{"q": "是的。上下水管都好的", "a": ["那就要检查线路了，一般风扇继电器是由电脑控制吸合的，如果电路存在断路，或者电脑坏了的话会出现继电器不吸合的情况！", "您好，先确定一下水循环是否正常，节温器有没有正常打开！如果都是好的，要检查风扇插头有没有电过来，如果有证明风扇是坏的，如果没有在继续查线路吧！"]}]}

### Steps To Reproduce

多轮对话按照(q1,a1),(q1+a1+q2,a2),(q1+a1+q2+a2+q3,a3)……是这样吗？
q和a的不变吗？
{
"id": 0, "paragraph": [
#一轮会话
{
"q"（还是q1）: "你好，我问一下，从南京到上海的路线是什么？",
"a": [
"你好，南京到上海的路线我不知道！" ]
},
{
"q"（还是q2）: "你好，我问一下，从南京到上海的路线是什么？你好，南京到上海的路线我不知道！那么，上海到北京路线呢？",
"a": [
"你好，路线是。。。。。。这样的。"
]
},
{
"q"（还是q3）: "你好，我问一下，从南京到上海的路线是什么？你好，南京到上海的路线我不知道！那么，上海到北京路线呢？你好，路线是。。。。。。这样的。好的，谢谢。",
"a": [
"不客气"
]
}
]
}

还是这样 (q1,a1),(q2,a2),(q3,a3)

{
"id": 0, "paragraph": [
#一轮会话
{
"q": "你好，我问一下，从南京到上海的路线是什么？",
"a": [
"你好，南京到上海的路线我不知道！" ]
},
{
"q": "那么，上海到北京路线呢？",
"a": [
"你好，路线是。。。。。。这样的。"
]
},
{
"q": "好的，谢谢。",
"a": [
"不客气"
]
}
]
}

### Environment

```markdown
- OS:
- Python:
- Transformers:
- PyTorch:
- CUDA Support (`python -c "import torch; print(torch.cuda.is_available())"`) :
```

**Comment by yuye2133** [comment]
楼主有答案了吗，同问，另外一个多轮对话里面的数据怎么组织进入模型训练啊，如果做成多个训练数据，shuffle不就打乱了么

**Comment by JaheimLee** [comment]
> 楼主有答案了吗，同问，另外一个多轮对话里面的数据怎么组织进入模型训练啊，如果做成多个训练数据，shuffle不就打乱了么

如果你的数据是按照`(q1,a1),(q1+a1+q2,a2),(q1+a1+q2+a2+q3,a3)……`组织的，shuffle打乱也没有影响啊

**Comment by pong991** [comment]
> ### Is there an existing issue for this?
> * [x]  I have searched the existing issues
> 
> ### Current Behavior
> 你好，如果领域多轮对话数据的话，如何构建多轮qa 数据示例进行微调？
> 
> 比如：我有个三轮的对话数据，下面这样构建微调数据集合理吗？q：代表用户说的话，a：代表机器说的话。 {"id": 1, "paragraph": [ {"q": "长城h3风扇不转。继电器好的。保险丝好的传感器新的风扇也新的这是为什么。就是继电器缺一个信号线", "a": ["用电脑能读数据流吗？水温多少"]} ,{"q": "95", "a": ["继续发动水温就高了？", "上下水管温差怎么样啊？空气是不是都排干净了呢？"]} ,{"q": "是的。上下水管都好的", "a": ["那就要检查线路了，一般风扇继电器是由电脑控制吸合的，如果电路存在断路，或者电脑坏了的话会出现继电器不吸合的情况！", "您好，先确定一下水循环是否正常，节温器有没有正常打开！如果都是好的，要检查风扇插头有没有电过来，如果有证明风扇是坏的，如果没有在继续查线路吧！"]}]}
> 
> ### Expected Behavior
> _No response_
> 
> ### Steps To Reproduce
> 多轮对话按照(q1,a1),(q1+a1+q2,a2),(q1+a1+q2+a2+q3,a3)……是这样吗？ q和a的不变吗？ { "id": 0, "paragraph": [ #一轮会话 { "q"（还是q1）: "你好，我问一下，从南京到上海的路线是什么？", "a": [ "你好，南京到上海的路线我不知道！" ] }, { "q"（还是q2）: "你好，我问一下，从南京到上海的路线是什么？你好，南京到上海的路线我不知道！那么，上海到北京路线呢？", "a": [ "你好，路线是。。。。。。这样的。" ] }, { "q"（还是q3）: "你好，我问一下，从南京到上海的路线是什么？你好，南京到上海的路线我不知道！那么，上海到北京路线呢？你好，路线是。。。。。。这样的。好的，谢谢。", "a": [ "不客气" ] } ] }
> 
> 还是这样 (q1,a1),(q2,a2),(q3,a3)
> 
> { "id": 0, "paragraph": [ #一轮会话 { "q": "你好，我问一下，从南京到上海的路线是什么？", "a": [ "你好，南京到上海的路线我不知道！" ] }, { "q": "那么，上海到北京路线呢？", "a": [ "你好，路线是。。。。。。这样的。" ] }, { "q": "好的，谢谢。", "a": [ "不客气" ] } ] }
> 
> ### Environment
> ```
> - OS:
> - Python:
> - Transformers:
> - PyTorch:
> - CUDA Support (`python -c "import torch; print(torch.cuda.is_available())"`) :
> ```
> 
> ### Anything else?
> _No response_
你好，后面选的哪种形式呢？
